src/pages/news/news-details/news-details.js

```diff
@@ -65,226 +89,154 @@ const NewsDetails = ({ match }) => {
     ukSetTitle,
     enSetAuthor,
     enSetText,
-    enSetTitle
+    enSetTitle,
+    setPreferredLanguages
   ]);
 
-  const newsSaveHandler = async (e) => {
-    e.preventDefault();
-    const newArticle = {
-      author: {
-        name: [
-          {
-            lang: languages[0],
-            value: ukAuthorName
-          },
-          {
-            lang: languages[1],
-            value: enAuthorName
-          }
-        ],
-        image: {
-          small: authorPhoto
-        }
-      },
-      title: [
-        {
-          lang: languages[0],
-          value: ukTitle
-        },
-        {
-          lang: languages[1],
-          value: enTitle
-        }
-      ],
-      text: [
-        {
-          lang: languages[0],
-          value: ukText
-        },
-        {
-          lang: languages[1],
-          value: enText
-        }
-      ],
-      images: {
-        primary: {
-          medium: newsImage
-        }
+  useEffect(() => {
+    const prefLanguages = [];
+    Object.keys(checkboxes).forEach((key) => {
+      if (checkboxes[key] === true) {
+        prefLanguages.push(key);
       }
-    };
-    dispatch(updateArticle({ id, newArticle }));
-  };
+    });
+    setPreferredLanguages(prefLanguages);
+  }, [checkboxes, setPreferredLanguages]);
 
+  const LanguageTabs =
```

Comment:
Please, lowercase the name of the variable. the uppercased name should have only functional components

Reply:
Fixed!